src/cfnlint/rules/resources/LimitNumber.py

```diff
@@ -17,13 +17,9 @@ class LimitNumber(CloudFormationLintRule):
 
     def match(self, cfn):
         """Check CloudFormation Resources"""
-
         matches = []
-
-        # Check number of resources against the defined limit
         resources = cfn.get_resources()
```

Comment:
wary about switching these from [`get_resources()`](https://github.com/aws-cloudformation/cfn-python-lint/blob/b34913e1186edcf23fd4042d2d197b76c762d708/src/cfnlint/template.py#L63) to [`get('Resources', {})`](https://github.com/aws-cloudformation/cfn-python-lint/blob/b34913e1186edcf23fd4042d2d197b76c762d708/src/cfnlint/decode/node.py#L108) like the rest of these rules

Reply:
I like `get_resources()` for these scenarios.  The idea being that we are validating that the basic construct of the resource is valid.  Its of type 'dict' for instance.  Ideally, that function could be built out even more.  Implement caching so we aren't checking that every time. Also check that Type is there no matter what. 

https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/template.py#L63-L81

Reply:
@kddejong [got these last couple rules refactored using `get_resources()` here](https://github.com/aws-cloudformation/cfn-python-lint/pull/1595)